README.md

```diff
@@ -30,90 +30,202 @@ When the reader has completed this code pattern, they will understand how to:
+source cfcreds.env
+mqtt_pub -i "d:${IOT_ORG_ID}:${IOT_DEVICE_TYPE}:${IOT_DEVICE_ID}" -u 'use-token-auth' -P "${IOT_DEVICE_AUTH_TOKEN}" -h "${IOT_ORG_ID}.messaging.internetofthings.ibmcloud.com" -p 1883 -t 'iot-2/evt/fromClient/fmt/json' -m '{
+    "d" : {
+          "sourceLanguage" : "en",
+          "payload" : "test",
+	        "client": "client1"
+    }
+}'
+```
+
+To create a set of credentials for subscribing to the broker, we'll need to go back to the IoT Platform dashboard and select the "Apps" option in the left hand menu. Then, click the "Generate API Key" button
+<p align="center">
+<img src="https://i.imgur.com/b9Iu9DS.png" width="650" height="450">
+</p>
+
+We can leave the fields in the "Information" blank and click next.
+In the "Permissions" tab, we'll select the "Backend Trusted Application" role. Once this is selected, click "Generate Key"
+<p align="center">
+<img src="https://i.imgur.com/ss6jpOZ.png" width="650" height="450">
+</p>
+
+The result will give us an API Key and Authentication Token. These can be supplied as the username and password for a MQTT client. To make setup a bit easier, place these values in the `cfcreds.env` file as IOT_API_KEY and IOT_APP_AUTH_TOKEN
+<p align="center">
+<img src="https://i.imgur.com/hfnB1B8.png" width="650" height="450">
+</p>
+
+Then we can subscribe to MQTT messages with
+```
+source cfcreds.env
+mqtt_sub -i "a:${IOT_ORG_ID}:test" -u "${IOT_API_KEY}" -P "${IOT_APP_AUTH_TOKEN}" -h "${IOT_ORG_ID}.messaging.internetofthings.ibmcloud.com" -p 1883 -t "iot-2/type/${IOT_DEVICE_TYPE}/id/${IOT_DEVICE_ID}/evt/toClients/fmt/json"
+
+```
+
+### 2. Deploy MQTT Feed
+
+Install the MQTT package/feed found in the openwhisk-package-mqtt-watson submodule [here](openwhisk-package-mqtt-watson). This "feed" enables Openwhisk to subscribe to one or more MQTT topics and invoke actions in response to incoming messages. To see more on how feeds work with IBM Cloud Functions, please visit these [documents](https://github.com/apache/incubator-openwhisk/blob/master/docs/feeds.md)
+
+### 3. Upload Actions
+Upload each "Action" to the Cloud Functions codebase with the following commands.
 ```
 bx wsk action create translateText translateText.js
 bx wsk action create sendSMS sendSMS.js
 bx wsk action create iotPub iotPub.py
-bx wsk action create handleIncomingSMS handleIncomingSMS.py
+bx wsk action create handleIncomingSMS handleIncomingSMS.js
 ```
 
-After each action is created, set default credentials for the corresponding services.
+After each action is created, set or bind default credentials for the corresponding services.
 ```
-# bx wsk action update getTTSToken --param TTS_PASSWD <passwd> --param TTS_USERNAME <username>
-bx wsk action update translateText --param language_translation_username ${language_translation_username} --param language_translation_password ${language_translation_password}
-...
+<!-- # bx wsk action update getTTSToken --param TTS_PASSWD <passwd> --param TTS_USERNAME <username> -->
```

Comment:
you should remove this comment or if you really want to keep it, move it out of the code block because it is visible